lib/semantics/check-omp-structure.h

```diff
@@ -147,6 +149,25 @@ class OmpStructureChecker : public virtual BaseChecker {
     return nullptr;
   }
 
+  template<typename T> std::optional<std::int64_t> GetIntValue(const T &x) {
+    const auto *expr{GetExpr(x)};
+    if (expr && evaluate::ToInt64(*expr).has_value()) {
+      return evaluate::ToInt64(*expr).value();
+    }
+    return std::nullopt;
+  }
+  template<typename T> std::int64_t GetPosIntValue(const T &x) {
+    const auto &v{GetIntValue(x)};
```

Comment:
I'm suprised this compiles, if it does.  Your `GetIntValue` routine doesn't return a reference, and I don't think that you can compare an `optional`.

The name of the function should make it very clear that it forces negative values to zero.

Reply:
Thanks Peter, I will fix the reference part. In terms of the function, do you have a suggestion? @klausler

Reply:
I would consider returning `std::optional<std::int64_t>` instead and leaving the name alone.

Reply:
Okay!

Reply:
Addressed in commit [8aa3737](https://github.com/flang-compiler/f18/pull/540/commits/8aa373705b1f740ddf49345affc5a1fa0bbd84d6). Thanks.